conda_build/utils.py

```diff
@@ -783,32 +783,14 @@ def _tar_xf_fallback(tarball, dir_path, mode="r:*"):
     from .os_utils.external import find_executable
 
     if tarball.lower().endswith(".tar.z"):
-        uncompress = find_executable("uncompress")
+        uncompress = find_executable("uncompress") or find_executable("gunzip")
         if not uncompress:
-            uncompress = find_executable("gunzip")
-        if not uncompress:
-            sys.exit(
-                """\
-uncompress (or gunzip) is required to unarchive .z source files.
-"""
-            )
+            sys.exit("uncompress/gunzip is required to unarchive .z source files.")
         check_call_env([uncompress, "-f", tarball])
         tarball = tarball[:-2]
 
-    t = tarfile.open(tarball, mode)
-    members = t.getmembers()
-    for i, member in enumerate(members, 0):
-        if os.path.isabs(member.name):
-            member.name = os.path.relpath(member.name, "/")
-        cwd = os.path.realpath(os.getcwd())
-        if not os.path.realpath(member.name).startswith(cwd):
-            member.name = member.name.replace("../", "")
-        if not os.path.realpath(member.name).startswith(cwd):
-            sys.exit("tarball contains unsafe path: " + member.name + " cwd is: " + cwd)
-        members[i] = member
-
-    t.extractall(path=dir_path)
-    t.close()
+    with tarfile.open(tarball, mode) as tar:
+        tar.extractall(path=dir_path, filter="data")
```

Comment:
This is used in a variety of locations (like in skeleton's unpacking of PyPI packages). Filtering for absolute paths outside of the destination is valid (the old code also did this with some homegrown logic). The question is whether it is also safe to remove group and other's write and execute bits.

Reply:
The key ideas are for tar logic are in [this conda-package-streaming subclass](https://github.com/conda/conda-package-streaming/blob/168beb4da1c971e8c99cee2c06c4862e267f8d81/conda_package_streaming/package_streaming.py#L38), I think. Looks like we don't care about owners, but we do want to respect the permissions as long as they conform to the umask.

Reply:
This should probably also be the `'tar'` filter.

Reply:
this thread is moot since we delegate to CPH/CPS